testing/src/main/java/org/oppia/android/testing/OppiaTestRule.kt

```diff
@@ -1,20 +1,45 @@
 package org.oppia.android.testing
 
 import android.os.Build
+import androidx.test.espresso.accessibility.AccessibilityChecks
+import androidx.test.espresso.matcher.ViewMatchers.withClassName
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import com.google.android.apps.common.testing.accessibility.framework.AccessibilityCheckResultUtils.matchesCheckNames
+import com.google.android.apps.common.testing.accessibility.framework.AccessibilityCheckResultUtils.matchesViews
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.endsWith
 import org.junit.AssumptionViolatedException
 import org.junit.rules.TestRule
 import org.junit.runner.Description
 import org.junit.runners.model.Statement
 
+private const val DEFAULT_ENABLED_STATE = true
+
 /** JUnit rule to enable [RunOn] test targeting. */
 class OppiaTestRule : TestRule {
   override fun apply(base: Statement?, description: Description?): Statement {
     return object : Statement() {
       override fun evaluate() {
+        val isEnabled = description.areAccessibilityChecksEnabled()
         val targetPlatforms = description.getTargetPlatforms()
         val targetEnvironments = description.getTargetEnvironments()
         val currentPlatform = getCurrentPlatform()
         val currentEnvironment = getCurrentBuildEnvironment()
+        if (currentPlatform == TestPlatform.ESPRESSO && isEnabled) {
```

Comment:
This check should only occur when all other conditions are met (i.e. that the correct platform is actually running) and thus should be moved into the first case of the ``when`` below.

Reply:
I have tried to place this condition in the first case of when but it's not working.

Reply:
@vrajdesai78 can you clarify what you specifically tried (a Gist of the actual code can help), and what specifically 'didn't work'?

Reply:
As you said I have tried to place this condition inside the first case of when 
![image](https://user-images.githubusercontent.com/43074241/150481529-e3ac872f-f095-4319-8dd3-39bdd5162cc3.png)
but after removing DisableAccessibilitychecks test cases are passing instead of failing. So, I have said that after placing coding in first case of when it's not working.

Reply:
This is actually what I meant:

```kotlin
        when {
          currentPlatform in targetPlatforms && currentEnvironment in targetEnvironments -> {
            // Only run this test if it's targeting the current platform & environment.
            if (currentPlatform == TestPlatform.ESPRESSO && areAccessibilityChecksEnabled) {
              AccessibilityChecks.enable().apply {
                // Suppressing failures for all views which matches with below conditions as we do not
                // want to change the UI to pass these failures as it will change the expected behaviour
                // for learner.
                setSuppressingResultMatcher(
                  allOf(
                    matchesCheckNames(`is`("TouchTargetSizeViewCheck")),
                    matchesViews(withContentDescription("More options")),
                    matchesViews(withClassName(endsWith("OverflowMenuButton")))
                  )
                )
              }.setRunChecksFromRootView(true)
            }
            base?.evaluate()
          }
          currentPlatform !in targetPlatforms -> {
            // See https://github.com/junit-team/junit4/issues/116 for context.
            throw AssumptionViolatedException(
              "Test targeting ${targetPlatforms.toPluralPlatformDescription()} ignored on" +
                " $currentPlatform"
            )
          }
          currentEnvironment !in targetEnvironments -> {
            throw AssumptionViolatedException(
              "Test targeting ${targetEnvironments.toPluralEnvironmentDescription()} ignored on" +
                " $currentEnvironment"
            )
          }
          else -> throw AssertionError("Reached impossible state in test rule")
        }
      }
```

Reply:
Thanks, @BenHenning for correcting me. I thought that I have to place it as a separate condition but now understood my mistake.